router/src/main/java/io/micronaut/web/router/AnnotatedMethodRouteBuilder.java

```diff
@@ -60,142 +59,178 @@
                     LOG.debug("Created Route: {}", route);
                 }
+                if (method.booleanValue(Get.class, "headRoute").orElse(true)) {
+                    route = HEAD(resolveUri(bean, uri,
+                            method,
+                            uriNamingStrategy),
+                            bean,
+                            method).produces(produces);
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Created Route: {}", route);
+                    }
+                }
             }
         });
 
         httpMethodsHandlers.put(Post.class, (BeanDefinition bean, ExecutableMethod method) -> {
-            String uri = method.stringValue(HttpMethodMapping.class).orElse(UriMapping.DEFAULT_URI);
-            MediaType[] consumes = resolveConsumes(method);
-            MediaType[] produces = resolveProduces(method);
-            Route route = POST(resolveUri(bean, uri,
-                method,
-                uriNamingStrategy),
-                bean,
-                method);
-            route = route.consumes(consumes).produces(produces);
-            if (LOG.isDebugEnabled()) {
-                LOG.debug("Created Route: {}", route);
+            AnnotationValue<Post> annotation = method.getAnnotation(Post.class);
+            Set<String> uris = CollectionUtils.setOf(annotation.stringValues("uris"));
+            uris.add(method.stringValue(HttpMethodMapping.class).orElse(UriMapping.DEFAULT_URI));
+            for (String uri: uris) {
+                MediaType[] consumes = resolveConsumes(method);
+                MediaType[] produces = resolveProduces(method);
+                Route route = POST(resolveUri(bean, uri,
+                        method,
+                        uriNamingStrategy),
+                        bean,
+                        method);
+                route = route.consumes(consumes).produces(produces);
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Created Route: {}", route);
+                }
             }
         });
 
         httpMethodsHandlers.put(CustomHttpMethod.class, (BeanDefinition bean, ExecutableMethod method) -> {
-            String uri = method.stringValue(HttpMethodMapping.class).orElse(UriMapping.DEFAULT_URI);
-            MediaType[] consumes = resolveConsumes(method);
-            MediaType[] produces = resolveProduces(method);
-            String methodName = method.stringValue(CustomHttpMethod.class, "method").get();
-            Route route = buildBeanRoute(methodName, HttpMethod.CUSTOM, resolveUri(bean, uri,
-                    method,
-                    uriNamingStrategy),
-                    bean,
-                    method);
-            route = route.consumes(consumes).produces(produces);
+            AnnotationValue<CustomHttpMethod> annotation = method.getAnnotation(CustomHttpMethod.class);
+            Set<String> uris = CollectionUtils.setOf(annotation.stringValues("uris"));
```

Comment:
So I am wondering if a set makes sense here. It would mean the match order defined in the array is not respected or is random?